drivers/scale_by_host_template_id.go

```diff
@@ -0,0 +1,404 @@
+package drivers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	v1client "github.com/rancher/go-rancher/client"
+	"github.com/rancher/go-rancher/v2"
+	rConfig "github.com/rancher/webhook-service/config"
+	"github.com/rancher/webhook-service/model"
+)
+
+var reg = regexp.MustCompile("[0-9]+$")
+
+type ScaleByHostTemplateIDDriver struct {
+}
+
+func (s *ScaleByHostTemplateIDDriver) ValidatePayload(conf interface{}, apiClient *client.RancherClient) (int, error) {
+	config, ok := conf.(model.ScaleByHostTemplateID)
+	if !ok {
+		return http.StatusInternalServerError, fmt.Errorf("Can't process config")
+	}
+
+	if config.Action == "" {
+		return http.StatusBadRequest, fmt.Errorf("Scale action not provided")
+	}
+
+	if config.Action != "up" && config.Action != "down" {
+		return http.StatusBadRequest, fmt.Errorf("Invalid action %v", config.Action)
+	}
+
+	if config.Amount <= 0 {
+		return http.StatusBadRequest, fmt.Errorf("Invalid amount: %v", config.Amount)
+	}
+
+	if config.HostTemplateID == "" {
+		return http.StatusBadRequest, fmt.Errorf("HostTemplateID not provided")
+	}
+
+	if config.Min <= 0 {
+		return http.StatusBadRequest, fmt.Errorf("Minimum scale not provided/invalid")
+	}
+
+	if config.Max <= 0 {
+		return http.StatusBadRequest, fmt.Errorf("Maximum scale not provided/invalid")
+	}
+
+	if config.Min >= config.Max {
+		return http.StatusBadRequest, fmt.Errorf("Max must be greater than min")
+	}
+
+	if config.Action == "up" {
+		if config.DeleteOption != "" {
+			return http.StatusBadRequest, fmt.Errorf("Delete option not to be provided while scaling up")
+		}
+	}
+
+	if config.Action == "down" {
+		if config.DeleteOption != "mostRecent" && config.DeleteOption != "leastRecent" {
+			return http.StatusBadRequest, fmt.Errorf("Invalid delete option/Delete option missing %v", config.DeleteOption)
+		}
+	}
+
+	return http.StatusOK, nil
+}
+
+func (s *ScaleByHostTemplateIDDriver) Execute(conf interface{}, apiClient *client.RancherClient, reqBody interface{}) (int, error) {
```

Comment:
This function needs to change significantly to only use the go-rancher client. You should not need to make any "raw" http calls anymore. You should be able to do everything you need to using the client.